caps-spark/src/test/scala/org/opencypher/caps/impl/spark/acceptance/ExpressionBehaviour.scala

```
@@ -615,6 +615,21 @@ trait ExpressionBehaviour {
           CypherMap("strings" -> Seq("bar", "foo"))
         ))
       }
+
+      it("can convert ListLiterals with nested non literal expressions") {
+        val graph = initGraph("CREATE ({val: 1}), ({val: 2})")
+
+        val result = graph.cypher(
+          """
+            |MATCH (n)
+            |WITH [n.val*10, n.val*100] as vals
+            |RETURN val""".stripMargin)
```

This doesn't look like valid Cypher. After the `WITH` horizon, only `vals` is available, so `RETURN val` should report an error; should be `RETURN vals`.